xbmc/utils/FileUtils.cpp

```diff
@@ -353,3 +354,19 @@ bool CFileUtils::Exists(const std::string& strFileName, bool bUseCache)
 {
   return CFile::Exists(strFileName, bUseCache);
 }
+
+bool CFileUtils::DeleteWithConfirm(const std::shared_ptr<CFileItem>& item)
+{
+  const bool confirm = CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
+      CSettings::SETTING_FILELISTS_CONFIRMFILEDELETE);
```

Comment:
@ksooo what worring me is that DeleteWithConfirm use CFileUtils that is generic, can be ok but
this setting should not be used for files that are not library related (audio/video/music) but nowhere is it specified, and no file type check happens here

Reply:
See my other comment. Don’t mix up usage and implantation. No checks for files belong here. This is controlled from outside, by the caller.

Reply:
if so, this code
```

  const bool confirm = CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
      CSettings::SETTING_FILELISTS_CONFIRMFILEDELETE);
```

is really wrong.
the `filelists.confirmfiledelete` is specified for audio/video/music and here IS NOT SO
then or you move this setting OUT of the "Media" to "System" and you made it generic for all kodi sections
or as is this is implemented is no good

Reply:
I agree, the setting belongs into another section of the settings, not system. At least for now.

Reply:
Actually this setting is meant to refer to the previous setting (allow file deletion). If the implementation of a generic method isn’t compatible with it, we should remove the generic method and implement it specifically into the locations where it is used. 
what cane be done is use the generic function and pass it the setting.
Personally, I don’t like the long line code of getting the setting, it’s not readable.
I don’t think this should be a generic for all files.  Everytime I make a new Kodi installation I change this setting and this feature was meant for that.

Reply:
Solved

Reply:
> what cane be done is use the generic function and pass it the setting.

NO! This is imo not the way to go.